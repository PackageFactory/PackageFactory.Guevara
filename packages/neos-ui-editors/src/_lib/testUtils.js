import {PureComponent, Children} from 'react';
import PropTypes from 'prop-types';
import {SynchronousRegistry, SynchronousMetaRegistry} from '@neos-project/neos-ui-extensibility/src/registry';

const globalRegistry = new SynchronousMetaRegistry();

// i18n Registry
class FakeI18NRegistry extends SynchronousRegistry {
    translate(key) {
        return key;
    }
}
globalRegistry.set('i18n', new FakeI18NRegistry());

// Data Source DataLoader
export const MockDataSourceDataLoader = {
    _currentPromise: null,
    _currentPromiseResolveFn: null,

    // API in tests
    reset() {
        this._currentPromise = null;
    },

    resolveCurrentPromise(result) {
        this._currentPromiseResolveFn(result);
        return this._currentPromise;
    },

    // API of DataLoader
    resolveValue() {
        if (!this._currentPromise) {
            this._currentPromise = new Promise(resolve => {
                this._currentPromiseResolveFn = resolve;
            });
        }
        return this._currentPromise;
    }
};
globalRegistry.set('dataLoaders', new SynchronousRegistry());
globalRegistry.get('dataLoaders').set('DataSources', MockDataSourceDataLoader);

// wrapper
export class WrapWithMockGlobalRegistry extends PureComponent {
    static propTypes = {
        children: PropTypes.element.isRequired
    };

    static childContextTypes = {
        globalRegistry: PropTypes.object.isRequired,
        configuration: PropTypes.object.isRequired
    };

    getChildContext() {
        const configuration = {};
        return {globalRegistry, configuration};
    }

    render() {
        return Children.only(this.props.children);
    }
}
