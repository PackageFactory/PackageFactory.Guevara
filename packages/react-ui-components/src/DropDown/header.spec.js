import sinon from 'sinon';
import {createShallowRenderer, createStubComponent} from './../_lib/testUtils.js';
import ShallowDropDownHeader from './header.js';

const Icon = createStubComponent();
const defaultProps = {
    children: 'Foo children',
    IconComponent: Icon,
    isOpen: false,
    toggleDropDown: () => null,
    theme: {/* eslint-disable quote-props */
        'dropDown__btn': 'baseDropDownHeaderClassName',
        'dropDown__chevron': 'baseDropDownHeaderChevronClassName'
    }/* eslint-enable quote-props */
};
const shallow = createShallowRenderer(ShallowDropDownHeader, defaultProps);

test('should render the "dropDown__btn" prop of the theme.', () => {
    const header = shallow();

    expect(header.hasClass('baseDropDownHeaderClassName')).toBeTruthy();
});
test('should render the "className" prop if passed.', () => {
    const header = shallow({
        className: 'barClassName'
    });

    expect(header.hasClass('barClassName')).toBeTruthy();
});
test('should call the "toggleDropDown" prop when clicking on the wrapper.', () => {
    const props = {
        toggleDropDown: sinon.spy()
    };
    const header = shallow(props);

    header.simulate('click');

    expect(props.toggleDropDown.calledOnce).toBeTruthy();
});
test('should call the "_refHandler" prop with the current "isOpen" prop when rendering the node.', () => {
    const props = {
        _refHandler: sinon.spy(),
        isOpen: 'foo'
    };
    shallow(props);

    expect(props._refHandler.calledOnce).toBeTruthy();
    expect(props._refHandler.args[0][0]).toBe('foo');
});
test('should render a node with a aria-haspopup attribute.', () => {
    const props = {
        _refHandler: sinon.spy(),
        isOpen: 'foo'
    };
    const header = shallow(props);

    expect(header.html().includes('aria-haspopup')).toBeTruthy();
});
test('should propagate the rest of the passed props to the wrapping node.', () => {
    const header = shallow({
        id: 'fooId'
    });

    expect(header.html().includes('id="fooId"')).toBeTruthy();
});
test('should render the passed "IconComponent" with the themes "dropDown__chevron" className and a "chevron-down" icon prop.', () => {
    const header = shallow();
    const icon = header.find(Icon);

    expect(icon.hasClass('baseDropDownHeaderChevronClassName')).toBeTruthy();
    expect(icon.prop('icon')).toBe('chevron-down');
});
test('should render the passed "IconComponent" with a "chevron-up" icon prop in case the "isOpen" prop is truthy.', () => {
    const header = shallow({isOpen: true});
    const icon = header.find(Icon);

    expect(icon.hasClass('baseDropDownHeaderChevronClassName')).toBeTruthy();
    expect(icon.prop('icon')).toBe('chevron-up');
});
