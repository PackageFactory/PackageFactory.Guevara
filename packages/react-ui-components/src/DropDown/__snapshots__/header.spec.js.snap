// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShallowDropDownHeader/> should render correctly. 1`] = `
<div
  aria-haspopup="true"
  className="baseDropDownHeaderClassName baseDropDownHeaderWithChevronClassName"
  onClick={[Function]}
  role="button"
>
  Foo children
  <Component
    className="baseDropDownHeaderChevronClassName"
    icon="chevron-down"
  />
</div>
`;
